unit/testing-utils/require_goto_statements.cpp

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************\
+
+ Module: Unit test utilities
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include "require_goto_statements.h"
+#include "catch.hpp"
+
+#include <algorithm>
+#include <util/expr_iterator.h>
+
+/// Expand value of a function to include all child codets
+/// \param instructions: The value of the function (e.g. got by looking up
+///  the function in the symbol table and getting the value)
+/// \return: All ID_code statements in the tree rooted at \p function_value
+std::vector<codet> require_goto_statements::get_all_statements(
+  const exprt &function_value)
+{
+  std::vector<codet> statements;
+
+  // Add any child statements (e.g. if this is a code_block
+  // TODO(tkiley): It should be possible to have a custom version of
```

Comment:
This is nasty enough to do with `std` algorithms, I'd just use the for-loop version if I were you.

Reply:
And just remove the TODO?

Reply:
You're quite right, for loop looks much simpler